crates/cargo-contract/src/cmd/config.rs

```diff
@@ -143,73 +138,75 @@ impl SignerConfig<Self> for Polkadot {
 
 /// Struct representing the implementation of the sr25519 signer
 #[derive(Clone)]
-pub struct SignerSR25519<C: Config>(pub PairSigner<C, sp_core::sr25519::Pair>);
+pub struct SignerSR25519<C: Config> {
+    account_id: <C as Config>::AccountId,
+    signer: sp_core::sr25519::Pair,
+}
 
-impl<C: Config> FromStr for SignerSR25519<C>
-where
-    <C as Config>::AccountId: From<sp_core::crypto::AccountId32>,
-{
+impl<C: Config<AccountId: From<AccountId32>>> SignerSR25519<C> {
+    /// Creates a new [`SignerSR25519`] from a [`sp_core::sr25519::Pair`].
+    pub fn new(signer: sp_core::sr25519::Pair) -> Self {
+        Self {
+            account_id: AccountId32(signer.public().0).into(),
+            signer,
+        }
+    }
+}
+
+impl<C: Config<AccountId: From<AccountId32>>> FromStr for SignerSR25519<C> {
     type Err = anyhow::Error;
 
     /// Attempts to parse the Signer suri string
     fn from_str(input: &str) -> Result<SignerSR25519<C>, Self::Err> {
         let keypair = sp_core::sr25519::Pair::from_string(input, None)?;
-        let signer = PairSigner::<C, _>::new(keypair);
-        Ok(Self(signer))
+        Ok(Self::new(keypair))
     }
 }
 
-impl<C: Config> SignerT<C> for SignerSR25519<C>
-where
-    <C as Config>::Signature: From<sp_core::sr25519::Signature>,
-{
+impl<C: Config<Signature: From<MultiSignature>>> SignerT<C> for SignerSR25519<C> {
     fn account_id(&self) -> <C as Config>::AccountId {
-        self.0.account_id().clone()
-    }
-
-    fn address(&self) -> C::Address {
-        self.0.address()
+        self.account_id.clone()
     }
 
     fn sign(&self, signer_payload: &[u8]) -> C::Signature {
-        self.0.sign(signer_payload)
+        MultiSignature::Sr25519(self.signer.sign(signer_payload).0).into()
     }
 }
 
 /// Struct representing the implementation of the ecdsa signer
 #[derive(Clone)]
-pub struct SignerEcdsa<C: Config>(pub PairSigner<C, sp_core::ecdsa::Pair>);
+pub struct SignerEcdsa<C: Config> {
+    account_id: <C as Config>::AccountId,
+    signer: sp_core::ecdsa::Pair,
+}
 
-impl<C: Config> FromStr for SignerEcdsa<C>
-where
-    // Requirements of the `PairSigner where:
-    // T::AccountId: From<SpAccountId32>`
-    <C as Config>::AccountId: From<sp_core::crypto::AccountId32>,
-{
+impl<C: Config<AccountId: From<AccountId32>>> SignerEcdsa<C> {
+    /// Creates a new [`SignerEcdsa`] from a [`sp_core::ecdsa::Pair`].
+    pub fn new(signer: sp_core::ecdsa::Pair) -> Self {
+        Self {
+            account_id: AccountId32(sp_core::blake2_256(signer.public().as_ref())).into(),
```

Comment:
Uses the same implementation from https://github.com/paritytech/polkadot-sdk/blob/c4ba6e0288a127f7f452e09e872960837aa313c0/substrate/primitives/runtime/src/lib.rs#L399, but without taking a dependency for a single conversion function.